scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -260,6 +260,14 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
                             "and $scannerCriteria, not scanning the package again $packageIndex."
                 }
 
+                storedResults.forEach {
+                    if (!archiver.hasArchive("${pkg.id.toPath()}/${it.provenance.hash()}")) {
```

Comment:
At the moment, I am a bit unsure about the case when the `storedResults` have more than one entry. The current code would download the same(?) source code each time and archive the result for each provenance. The download should obviously be called only once, that I need to change. But does archiving the same results for each provenance make sense?

It seems that the provenance  depends on download date and VCS data. Does this mean that the corresponding hash depends on the state of the repository when the package was first downloaded and scanned? If so, should the current archive result be put together with an earlier scan result? I am thinking about packages which are continuously updated and have no real versioning.

Maybe someone can enlighten me.

Reply:
I didn't think of the case before that there could be multiple results for the same provenance, e.g. from different scanners. I think it would be better to first collect all `Provenance` objects in a set and then iterate over this to prevent downloading the same source multiple times.

Another case would be multiple results with different provenance, e.g. one for VCS and one for a source artifact. In this case it would be correct to download it multiple times, as the archived files could be different.